script.js

```diff
@@ -1,12 +1,96 @@
-//You can edit ALL of the code here
-function setup() {
+const setup = () => {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
-}
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
-}
+  const searchBox = document.getElementById("searchBox");
+  const searchCount = document.getElementById("searchCount");
+  searchCount.innerText = "search episodes";
+  //whenever we type a letter in a search box
+  searchBox.addEventListener("keyup", search);
+
+  const episodeSelection = document.getElementById("episodeSelection");
+  //whenever we change the dropdown list
+  episodeSelection.addEventListener("change", showSelectedEpisode);
+};
+
+//function will call when user change the dropdown list value
+const showSelectedEpisode = (event) => {
+  const episodeId = Number(event.target.value);
```

Comment:
Hi Laleh, great job, I think target.value is type of number. Is there any reason to change it to number?

Reply:
@Hatice06 thanks a lot Hatice for reviewing my code, when I debugged my code understand that event.target.value returns string and when I want to compare it with episode.id that it is a number so it didnt work.

when I reviewed your code I understand you did same as me but without using Number function and It works for you . I think it's because you used (==) in your comparison and it compares two variables without considering their types and I used (===) for comparing my variables so I have to use Number function.

Reply:
You are right Laleh, I have used  ==  and then I didn't check the type.